include/stumpless/target/network.h

```diff
@@ -669,6 +669,54 @@ struct stumpless_target *
 stumpless_set_udp_max_message_size( struct stumpless_target *target,
                                     size_t max_msg_size );
 
+/**
+ * Gets the network protocol of a network target.
+ *
+ * **Thread Safety: MT-Safe**
+ * This function is thread safe. A mutex is used to coordinate the read of the
+ * target with other accesses and modifications.
+ *
+ * **Async Signal Safety: AS-Unsafe lock heap**
+ * This function is not safe to call from signal handlers due to the use of a
+ * non-reentrant lock to coordinate access.
+ *
+ * **Async Cancel Safety: AC-Unsafe lock heap**
+ * This function is not safe to call from threads that may be asynchronously
+ * cancelled, due to the use of a lock that could be left locked.
+ *
+ * @param target The target to get the network protocol from.
+ *
+ * @return The network protocol of the network target.
+ * In the event of an error, NULL is returned and an error code is 
+ * set appropriately.
+ */
+enum stumpless_network_protocol
```

Comment:
The `STUMPLESS_PUBLIC_FUNCTION` designator is missing here, which is causing the build errors you are seeing on Linux. On Windows, the functions need to be added to `src/windows/stumpless.def` to make them publicly visible.